typescript/react-navigation.d.ts

```diff
@@ -549,14 +549,16 @@ declare module 'react-navigation' {
   export interface NavigationTabRouterConfig
     extends NavigationTabRouterConfigBase {
     defaultNavigationOptions?: NavigationScreenConfig<NavigationScreenOptions>;
-    navigationOptions?: NavigationTabRouterConfigBase;
+    navigationOptions?: NavigationScreenConfig<NavigationScreenOptions>;
   }
   export interface NavigationBottomTabRouterConfig
     extends NavigationTabRouterConfigBase {
     defaultNavigationOptions?: NavigationScreenConfig<
       NavigationBottomTabScreenOptions
     >;
-    navigationOptions?: NavigationTabRouterConfigBase;
+    navigationOptions?: NavigationScreenConfig<
+      NavigationBottomTabScreenOptions
```

Comment:
we can't actually type this very well because the type depends on the parent navigator :\ 

```js
let Tabs = createBottomTabNavigator({ Home }, {
  defaultNavigationOptions: { /* this applies to Home route to configure bottom tabs */ },
  navigationOptions: { /* this applies to Tabs route on stack to configure stack */ },
});
let Stack = createStackNavigator({ Tabs });
```

might have to do `NavigationScreenConfig<any>` or `NavigationScreenConfig< NavigationScreenOptions>` for now. of course this isn't great either because `NavigationScreenOptions` is only a union of build in navigation option types.. so probably `any` is the most accurate

Reply:
Hmmm, makes sense i guess it is the best option.